addons/populate/functions/fnc_spawnBuildingUnit.sqf

```diff
@@ -0,0 +1,95 @@
+#include "script_component.hpp"
+/*
+    Author:
+        Bridg
+
+    Description:
+        Spawns a unit at the passed position
+
+    Parameters:
+        0: _spawnPositions <ARRAY>
+        1: _numberOfUnitsToSpawn <NUMBER>
+        2: _numberOfPositionsToOccupy <NUMBER>
+        3: _side <SIDE>
+        4: _module <MODULE>
+        5: _unitPoolArray <ARRAY>
+        6: _statics <ARRAY>
+
+    Return value:
+        Nothing
+*/
+
+params ["_spawnPositions", "_numberOfUnitsToSpawn", "_numberOfPositionsToOccupy", "_side", "_module", "_unitPoolArray", "_statics"];
+
+private _group = grpNull;
+private _currenGrouptUnitCount = 0;
+
+[{
+    params ["_args", "_idPFH"];
+    _args params ["_spawnPositions", "_numberOfUnitsToSpawn", "_numberOfPositionsToOccupy", "_side", "_module", "_unitPoolArray", "_currenGrouptUnitCount", "_group", "_statics"];
+
+    private _pos = [0,0,0];
+    private _dir = -1;
+
+    if (_numberOfPositionsToOccupy <= 1) exitWith {
+        [_idPFH] call cba_fnc_removePerFrameHandler;
+        if (_numberOfUnitsToSpawn > 0) then {
+            [_numberOfUnitsToSpawn, _side, _module, _unitPoolArray] call FUNC(createPatrols);
+        ;}
+    };
+
+    // create a group if the _currenGrouptUnitCount is == 0
+    if (_currenGrouptUnitCount isEqualTo 0) then {
+        _group = createGroup _side;
+    };
+
+    private _unitType = selectRandom _unitPoolArray;
+    private _unit = _group createUnit [_unitType, [0,0,0], [], 0, "NONE"];
+
+    _currenGrouptUnitCount = _currenGrouptUnitCount + 1;
+    _numberOfUnitsToSpawn = _numberOfUnitsToSpawn - 1;
+
+    // do statics first - use exit with to stop spawning of other units
+    if (_statics isNotEqualTo []) exitWith {
+        private _static = selectRandom _statics;
+        _unit moveInAny _static;
+        _statics deleteAt (_statics findIf {_x isEqualTo _static});
+        _numberOfPositionsToOccupy = _numberOfPositionsToOccupy - 1;
+
+        _args set [1, _numberOfUnitsToSpawn];
+        _args set [2, _numberOfPositionsToOccupy];
+        _args set [6, _currenGrouptUnitCount];
+        _args set [7, _group];
+        _args set [8, _statics];
```

Comment:
Statics is an array, don't need to update it if elements are being removed (variable is a reference)

Reply:
did not know it was clever enough to do that, thought you had to set it back because of scoping